src/regex.c

```diff
@@ -286,19 +287,22 @@ int do_regex_hybrid_crack(struct db_main *db, const char *regex,
 	}
 
 	while (c_iterator_next(iter)) {
-		c_iterator_value(iter, buffer);
-		c_simplestring_to_utf8_string(buffer, &word[0], sizeof(word));
 		c_simplestring_clear(buffer);
-		if (options.internal_cp != UTF_8)
-			utf8_to_cp_r(word, word, sizeof(word));
+		c_iterator_value(iter, buffer);
+		word = c_simplestring_to_string(buffer);
+    /**
+      * rexgen already creates the correct encoding
+      */
+		//if (options.internal_cp != UTF_8)
+		//	utf8_to_cp_r(word, word, sizeof(word));
 		if (options.mask) {
```

Comment:
I'm using the currently configured locale as default for input as well as output. I suppose that I need to implement a way to explicitly define the encodings, because e.g. a file from which I read my have a different encoding than the current terminal has. I'm not yet sure how I'll do this.